src/dataframerow/utils.jl

```diff
@@ -94,6 +94,43 @@ isequal_row(cols1::Tuple{Vararg{AbstractVector}}, r1::Int,
     isequal(cols1[1][r1], cols2[1][r2]) &&
         isequal_row(Base.tail(cols1), r1, Base.tail(cols2), r2)
 
+# Simple vector type for internal use which represents a virtual DataAPI.refpool
+# so that an integer vector can be its own DataAPI.refarray:
+# this just allows telling the generic row_group_slots method for generic refpools
+# what is the minimum index and the number of (potential) groups
+struct IntegerRefpool{T} <: AbstractVector{T}
+    min::Int
+    max::Int
+end
+
+Base.size(x::IntegerRefpool) = (x.max - x.min + 1,)
+Base.axes(x::IntegerRefpool) = (x.min:x.max,)
+Base.IndexStyle(::Type{<:IntegerRefpool}) = Base.IndexLinear()
+@inline function Base.getindex(x::IntegerRefpool, i::Integer)
+    @boundscheck checkbounds(x, i)
+    return i - x.min + 1
+end
+Base.allunique(::IntegerRefpool) = true
+Base.issorted(::IntegerRefpool) = true
+
+function _refpool(x::AbstractArray)
+    refpool = DataAPI.refpool(x)
+    if refpool !== nothing
+        return refpool
+    elseif x isa AbstractArray{<:Real} && !isempty(x) && all(isinteger, x)
+        minval, maxval = extrema(x)
+        # Threshold chosen with the same rationale as the row_group_slots refpool method:
+        # refpool approach is faster but we should not allocate too much memory either
+        if maxval - minval + 1 <= 2 * length(x)
```

Comment:
It seems that even above this threshold the `refpool` approach is faster. But I'm not sure how much memory we're willing to allocate for speed.